omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java

```diff
@@ -232,20 +233,33 @@ protected PageableResult doSearch(RequestContext context) {
 			
 			String careSettingUuid = context.getRequest().getParameter("careSetting");
 			String asOfDateString = context.getRequest().getParameter("asOfDate");
+			String orderTypeUuid = context.getRequest().getParameter("orderType");
 			String sortParam = context.getRequest().getParameter("sort");
 			
 			CareSetting careSetting = null;
 			Date asOfDate = null;
+			OrderType orderType = null;
 			if (StringUtils.isNotBlank(asOfDateString)) {
 				asOfDate = (Date) ConversionUtil.convert(asOfDateString, Date.class);
 			}
 			if (StringUtils.isNotBlank(careSettingUuid)) {
 				careSetting = ((CareSettingResource1_10) Context.getService(RestService.class).getResourceBySupportedClass(
 				    CareSetting.class)).getByUniqueId(careSettingUuid);
+				if (careSetting == null) {
+					return new EmptySearchResult();
+				}
+			}
+			if (StringUtils.isNotBlank(orderTypeUuid)) {
+				orderType = ((OrderTypeResource1_10) Context.getService(RestService.class).getResourceBySupportedClass(
+				    OrderType.class)).getByUniqueId(orderTypeUuid);
+				if (orderType == null) {
```

Comment:
Why do you return empty result?

Reply:
Similar to the case with care settings, if an invalid order type is passed in, it should return an empty result rather than return all orders.

Reply:
See my comment to that, it's misleading to send back an empty list.

Reply:
@wluyima I had been working on this with @dsurrao, and we went back and forth on this a bit.  We agreed what throwing an error seemed like the best solution, but noticed there was precedence for just returning an an empty set earlier in the list:

https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java#L218

So we just went with that for consistent, but are fine with changing them to fail loudly... it looking at another file, it looks like throwing an IllegalArgumentExcepotion is the current thing to do?

Reply:
That existing code that returns an empty list is doing the wrong thing too.

Reply:
I think it's better to throw `ObjectNotFoundException`

Reply:
Thanks for the feedback! I've created a new PR with these updates.